### PR TITLE
MAT-7038: Modify cs retrieve to return displayValues for vsac version

### PR DESCRIPTION
**Pull request by mcmcphillips**
## Terminology Service PR

Jira Ticket: [MAT-7038](https://jira.cms.gov/browse/MAT-7038)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [x] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java [hunk 1]
```diff
@@ -122,7 +124,49 @@ private List<QdmValueSet.Concept> getValueSetConcepts(
   }
 
   public List<CodeSystem> getAllCodeSystems() {
-    return codeSystemRepository.findAll();
+    // remove items that are marked as not present in vsac to cut expense
+    List<CodeSystemEntry> codeSystemEntries = mappingService.getCodeSystemEntries().stream().filter(codeSystemEntry -> !codeSystemEntry.getOid().contains("NOT.IN.VSAC")).toList();
+    List<CodeSystem> codeSystems = codeSystemRepository.findAll();
+    List<CodeSystem> transformedResults = new ArrayList<>(Collections.emptyList());
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java]: Passing in the empty list to the ArrayList constructor doesn't seem necessary. 

From the [ArrayList javadoc](https://docs.oracle.com/en%2Fjava%2Fjavase%2F21%2Fdocs%2Fapi%2F%2F/java.base/java/util/ArrayList.html) on passing in a Collections to the constructor:
> Constructs a list containing the elements of the specified collection, in the order they are returned by the collection's iterator.

Thus this taking an empty list and turning it into an empty list by iterating over an empty list.

File: src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java [hunk 2]
```diff
@@ -122,7 +124,49 @@ private List<QdmValueSet.Concept> getValueSetConcepts(
   }
 
   public List<CodeSystem> getAllCodeSystems() {
-    return codeSystemRepository.findAll();
+    // remove items that are marked as not present in vsac to cut expense
+    List<CodeSystemEntry> codeSystemEntries = mappingService.getCodeSystemEntries().stream().filter(codeSystemEntry -> !codeSystemEntry.getOid().contains("NOT.IN.VSAC")).toList();
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java:128]: The mapping is only needed for QDM measures. We shouldn't be limiting the results from the DB based on the mapping doc for FHIR based measures.

**Review by jkotanchik-SB — APPROVED** [review]
